whylog/config/investigation_plan.py

```diff
@@ -48,9 +48,52 @@ def is_line_in_search_range(self, super_parser_groups):
     def compare_with_bound(self, bound, super_parser_groups):
         """
         Basing on super_parser_groups extracted from line, returns information
-        how relative to bound this line is.
+        how relative to choosed bound (LEFT_BOUND or RIGHT_BOUND) this line is.
+        Example:
+            bound = InvestigationStep.LEFT_BOUND (only other possible value of this parameter is
+                    InvestigationStep.RIGHT_BOUND)
+            super_parser_groups = [('date', datetime(2015, 12, 3, 12, 8, 9))]
+            self._search_ranges = {
+                'date': {
+                    'left_bound': datetime(2015, 12, 3, 12, 8, 0),
+                    'right_bound': datetime(2015, 12, 3, 12, 8, 11)
+                }
+            }
+            returned value: CompareResult.GT
+        When self._search_ranges hasn't subdictonary for given super_parser_groups's type or
+        super_parser_groups parameter is empty list return CompareResult.LT/CompareResult.GT
+        when compare with LEFT_BOUND/RIGHT_BOUND. This means that InvestigationStep object hasn't
+        information about order in parsed file.
         """
-        #TODO: remove mock
+        # This implementation assume that super_parser_groups length equals 1 or 0
+        # TODO implementation for longer super_parser_groups_list
+        group_value, bound_value = self._extract_values_to_compare(bound, super_parser_groups)
+        if bound_value is None:
+            return self._compare_with_undefined_bound(bound)
+        return self._compare_extracted_values(bound_value, group_value)
+
+    def _extract_values_to_compare(self, bound, super_parser_groups):
+        if not super_parser_groups:
+            return None, None
+        group_type, group_value = super_parser_groups[0]
+        # type_bound is a dictionary, that contains bounds (LEFT and RIGHT bound)
+        # values for concrete primary key type
+        type_bounds = self._search_ranges.get(group_type)
+        if type_bounds is None:
+            return None, None
+        return group_value, type_bounds[bound]
+
+    def _compare_with_undefined_bound(self, bound):
```

Comment:
No, it's means that investigation step hasn't defined bounds for this primary key type.